contracts/treasury/GovernorCountingOverridable.sol

```diff
@@ -181,13 +181,20 @@ abstract contract GovernorCountingOverridable is Initializable, GovernorUpgradea
         uint256 timepoint = proposalSnapshot(_proposalId);
         address delegate = votes().delegatedAt(_account, timepoint);
 
-        bool isTranscoder = _account == delegate;
-        if (isTranscoder) {
-            // deduce weight from any previous delegators for this transcoder to
-            // make a vote
+        bool selfDelegates = _account == delegate;
+        if (selfDelegates) {
+            // deduce weight from any previous delegators for this self-delegating account to cast a vote
             return _weight - _voter.deductions;
         }
 
+        address delegatesDelegate = votes().delegatedAt(delegate, timepoint);
+        bool delegateSelfDelegates = delegate == delegatesDelegate;
```

Comment:
IMO `isDelegateSelfDelegated` would be easier to read.

Reply:
Good call I struggled with this one 😅